initializr-generator/src/main/groovy/io/spring/initializr/metadata/InitializrConfiguration.groovy

```diff
@@ -123,6 +123,14 @@ class InitializrConfiguration {
 		String springBootMetadataUrl = 'https://spring.io/project_metadata/spring-boot'
 
 		/**
+		 * The group / artifact / version of a custom parent pom to use for generated projects.
+		 * This is only enabled if a value is expliclty provided.
+		 *
+		 * The value must be specified in "groupid:artifactId:versionNumber" format
+		 */
+		String customParentPomGAV
```

Comment:
I don't like the name and I think it should be some higher-level object, something like

```
env:
  parentPom:
      groupId: com.example
      artifactId: my-parent-pom
      includeSpringBootBom: false

With `ParentPom` being an inner class of `Env`. Please also make sure to update `spring-configuration-metadata`